src/provider.rs

```diff
@@ -172,6 +197,201 @@ impl TableProvider for SeafowlTable {
             inner: plan,
         }))
     }
+
+    fn supports_filter_pushdown(
+        &self,
+        _filter: &Expr,
+    ) -> Result<TableProviderFilterPushDown> {
+        Ok(TableProviderFilterPushDown::Inexact)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SeafowlPruningStatistics {
+    pub partition_count: usize,
+    pub schema: SchemaRef,
+    pub partitions: Arc<Vec<SeafowlPartition>>,
+    pub min_values: HashMap<String, Vec<ScalarValue>>,
+    pub max_values: HashMap<String, Vec<ScalarValue>>,
+    pub null_counts: HashMap<String, Vec<Option<u64>>>,
+}
+
+impl SeafowlPruningStatistics {
+    // Generate maps of min/max/null stats per column, parsing the serialized values stored in partitions
+    fn from_partitions(
+        partitions: Vec<SeafowlPartition>,
+        schema: SchemaRef,
+    ) -> Result<Self> {
+        let partition_count = partitions.len();
+        let mut min_values = HashMap::new();
+        let mut max_values = HashMap::new();
+        let mut null_counts = HashMap::new();
+
+        for field in schema.fields() {
+            min_values.insert(
+                field.name().clone(),
+                vec![ScalarValue::Utf8(None); partition_count],
+            );
+            max_values.insert(
+                field.name().clone(),
+                vec![ScalarValue::Utf8(None); partition_count],
```

Comment:
They should actually need to be `ScalarValue::Int16(None)`, `ScalarValue::Boolean(None)`, etc. depending on the type. But since we agreed on the JIT-like stats parsing I'll be changing that part anyway.